xen/arch/x86/boot/head.S

```diff
@@ -486,20 +486,34 @@ __start:
         jmp     trampoline_bios_setup
 
 .Lslaunch_proto:
+        /* Upon reaching here, CPU state mostly matches the one setup by the
+         * bootloader with ESP, ESI and EDX being clobbered above. */
+
         /* Save information that TrenchBoot slaunch was used. */
         movb    $1, sym_esi(slaunch_active)
 
+        /* Prepare space for output parameter of slaunch_early_tests(), which is
+         * a structure of two uint32_t fields. */
+        sub     $8, %esp
+
+        push    %esp                             /* pointer to output structure */
         lea     sym_offs(__2M_rwdata_end), %ecx  /* end of target image */
         lea     sym_offs(_start), %edx           /* target base address */
         mov     %esi, %eax                       /* load base address */
-        /* txt_early_tests(load/eax, tgt/edx, tgt_end/ecx) using fastcall. */
-        call    txt_early_tests
+        /* slaunch_early_tests(load/eax, tgt/edx, tgt_end/ecx, ret/stk) using fastcall. */
+        call    slaunch_early_tests
```

Comment:
Without `__stdcall` attribute on `slaunch_early_tests` this line must be followed by `add $4, %esp` to clean up the stack, which should be changed to `add $8, %esp` in the next commit (dad8074fc33891ad05f7073ac751de704c87791f).

Reply:
Right, this isn't true `__fastcall`:
https://github.com/TrenchBoot/xen/commit/b8aefd43c83f676ebf8e3f0405591ad8b5d0e2b5#diff-e4b4491b6c4e847098c559822abc0da4b62e431b9339a971a79026a0c3bac48aR511
https://github.com/TrenchBoot/xen/commit/d793572382cf94d64a25a4725370ea7f7439cc35#diff-e4b4491b6c4e847098c559822abc0da4b62e431b9339a971a79026a0c3bac48aR536